drivers/display/ek79652.c

```diff
@@ -0,0 +1,552 @@
+				  const enum display_orientation
+				  orientation)
+{
+	LOG_ERR("Unsupported");
+	return -ENOTSUP;
+}
+
+static int ek79652_set_pixel_format(const struct device *dev,
+				   const enum display_pixel_format pf)
+{
+	if (pf == PIXEL_FORMAT_MONO10) {
+		return 0;
+	}
+
+	LOG_ERR("not supported");
+	return -ENOTSUP;
+}
+
+static int ek79652_clear_display(const struct device *dev,
+					 uint8_t pattern, bool update)
+{
+	struct display_buffer_descriptor desc = {
+		.buf_size = EK79652_NUMOF_PAGES,
+		.width = EPD_PANEL_WIDTH,
+		.height = 1,
+		.pitch = EPD_PANEL_WIDTH,
+	};
+	uint8_t *line;
+
+	line = k_malloc(EK79652_NUMOF_PAGES);
+	if (line == NULL) {
+		return -ENOMEM;
+	}
+
+	memset(line, pattern, EK79652_NUMOF_PAGES);
+
+	for (int i = 0; i < EPD_PANEL_HEIGHT; i++) {
+		ek79652_write(dev, 0, i, &desc, line);
+	}
+
+	k_free(line);
+
+	if (update == false) {
+		if (ek79652_update_display(dev)) {
+			return -EIO;
+		}
+	}
+
+	return 0;
+}
+
+static int ek79652_controller_init(const struct device *dev)
+{
+	uint8_t tmp[EK79652_PWROPT_LENGTH];
+	struct ek79652_data *driver = dev->data;
+
+	gpio_pin_set(driver->reset, EK79652_RESET_PIN, 0);
+	k_sleep(K_MSEC(EK79652_RESET_DELAY));
+	gpio_pin_set(driver->reset, EK79652_RESET_PIN, 1);
+	k_sleep(K_MSEC(EK79652_RESET_DELAY));
+	gpio_pin_set(driver->reset, EK79652_RESET_PIN, 0);
+	k_sleep(K_MSEC(EK79652_RESET_DELAY));
+	ek79652_busy_wait(driver);
+
+	LOG_DBG("Initialize EK79652 controller");
+
+	if (ek79652_write_cmd(driver, EK79652_CMD_PWR, ek79652_pwr,
+			     sizeof(ek79652_pwr))) {
+		return -EIO;
+	}
+
+	if (ek79652_write_cmd(driver, EK79652_CMD_BTST, ek79652_softstart,
+			     sizeof(ek79652_softstart))) {
+		return -EIO;
+	}
+
+	/* Power optimization */
+#if DT_INST_NODE_HAS_PROP(0, pwropt)
+	tmp[0] = ek79652_pwropt[0];
+	tmp[1] = ek79652_pwropt[1];
+	if (ek79652_write_cmd(driver, EK79652_CMD_PWROPT, tmp,
+		EK79652_PWROPT_LENGTH)) {
+		return -EIO;
+	}
+
+	tmp[0] = ek79652_pwropt[2];
+	tmp[1] = ek79652_pwropt[3];
+	if (ek79652_write_cmd(driver, EK79652_CMD_PWROPT, tmp,
+		EK79652_PWROPT_LENGTH)) {
+		return -EIO;
+	}
+
+	tmp[0] = ek79652_pwropt[4];
+	tmp[1] = ek79652_pwropt[5];
+	if (ek79652_write_cmd(driver, EK79652_CMD_PWROPT, tmp,
+		EK79652_PWROPT_LENGTH)) {
+		return -EIO;
+	}
+
+	tmp[0] = ek79652_pwropt[6];
+	tmp[1] = ek79652_pwropt[7];
+	if (ek79652_write_cmd(driver, EK79652_CMD_PWROPT, tmp,
+		EK79652_PWROPT_LENGTH)) {
+		return -EIO;
+	}
+
+	tmp[0] = ek79652_pwropt[8];
+	tmp[1] = ek79652_pwropt[9];
+	if (ek79652_write_cmd(driver, EK79652_CMD_PWROPT, tmp,
+		EK79652_PWROPT_LENGTH)) {
+		return -EIO;
+	}
+
+	tmp[0] = ek79652_pwropt[10];
+	tmp[1] = ek79652_pwropt[11];
+	if (ek79652_write_cmd(driver, EK79652_CMD_PWROPT, tmp,
+		EK79652_PWROPT_LENGTH)) {
+		return -EIO;
+	}
+#endif
+
+	/* Reset DFV_EN */
+	tmp[0] = EK79652_PDRF_VAL;
+	if (ek79652_write_cmd(driver, EK79652_CMD_PDRF, tmp, 1)) {
+		return -EIO;
+	}
+
+	/* Turn on: booster, controller, regulators, and sensor. */
```

Comment:
Does it have a temperature sensor like gd7965 controller? Does this controller have valid waveforms in OTP ROM (If not, return it to gooddisplay)? Do you understand the relationship between temperature sensor, OTP ROM and waveforms? And that what you do in line 477 does not fit together?

Reply:
yes it does have. _set_partial_lut_ is effective only for partial display refresh.